packages/collector/test/tracing/database/couchbase/test.js

```diff
@@ -0,0 +1,674 @@
+            }));
+      }
+
+      if (apiType === 'promise') {
+        it('[datastructures list] must trace', () =>
+          controls
+            .sendRequest({
+              method: 'get',
+              path: `/datastructures-list-${apiType}`
+            })
+            .then(resp => {
+              expect(resp.iteratedItems).to.eql(['test1', 'test2']);
+
+              return retry(() =>
+                verifySpans(agentControls, controls, {
+                  spanLength: 9,
+                  verifyCustom: (entrySpan, spans) => {
+                    expectExactlyNMatching(
+                      spans,
+                      3,
+                      verifyCouchbaseSpan(controls, entrySpan, {
+                        sql: 'MUTATEIN'
+                      })
+                    );
+                    expectExactlyNMatching(
+                      spans,
+                      2,
+                      verifyCouchbaseSpan(controls, entrySpan, {
+                        sql: 'LOOKUPIN'
+                      })
+                    );
+                    expectExactlyNMatching(
+                      spans,
+                      3,
+                      verifyCouchbaseSpan(controls, entrySpan, {
+                        sql: 'GET'
+                      })
+                    );
+                  }
+                })
+              );
+            }));
+
+        it('[datastructures map] must trace', () =>
+          controls
+            .sendRequest({
+              method: 'get',
+              path: `/datastructures-map-${apiType}`
+            })
+            .then(resp => {
+              expect(resp.iteratedItems).to.eql(['test1', 'test2']);
+
+              return retry(() =>
+                verifySpans(agentControls, controls, {
+                  spanLength: 9,
+                  verifyCustom: (entrySpan, spans) => {
+                    expectExactlyNMatching(
+                      spans,
+                      3,
+                      verifyCouchbaseSpan(controls, entrySpan, {
+                        sql: 'MUTATEIN'
+                      })
+                    );
+                    expectExactlyNMatching(
+                      spans,
+                      3,
+                      verifyCouchbaseSpan(controls, entrySpan, {
+                        sql: 'LOOKUPIN'
+                      })
+                    );
+                    expectExactlyNMatching(
+                      spans,
+                      2,
+                      verifyCouchbaseSpan(controls, entrySpan, {
+                        sql: 'GET'
+                      })
+                    );
+                  }
+                })
+              );
+            }));
+      }
+
+      it('[error] must trace remove', () =>
+        controls
+          .sendRequest({
+            method: 'post',
+            path: `/remove-${apiType}?error=true`
+          })
+          .then(resp => {
+            expect(resp.success).to.eql(true);
```

Comment:
Related to the comment in the test app, we should let the test app return the error message and verify it here.

Reply:
I'll add it.

Reply:
I had this before. Now I remember why I have simply removed it. Both cases return a different err msg because I test two different cases.